Feature/teleop

bgrayd opened the pull request:
I have started the teleop node, but it still needs the service calls to services that haven't been written yet.  It also needs to do the arm control, but I am not sure how defined that is yet.  This should also be able to be used in gazebo.

bgrayd committed:
Created teleop node

bgrayd committed:
Forgot a few things...

JacobAMason commented on cr17/nodes/cr17/joystick_controller.py:
```diff
@@ -0,0 +1,156 @@
+        rospy.Subscriber(self.joystick_topic, Joy, self.joy_callback)
+
+    def startTeleop(self):
+        #TODO: call the serives from navigator, scoop_controller, and the HLSC
```
"serives" ?

JacobAMason commented on cr17/nodes/cr17/joystick_controller.py:
```diff
@@ -0,0 +1,156 @@
+        self.teleopEnabled = True
+
+    def endTeleop(self):
+        #TODO: call the serives from navigator, scoop_controller, and the HLSC
```
"serives"

JacobAMason commented on cr17/nodes/cr17/joystick_controller.py:
```diff
@@ -0,0 +1,156 @@
+            self.joy_received = False
+            rate.sleep()
+
+    def _exit_handler(self):
```
Is this method ever used?

JacobAMason reviewed (changes requested):
startTeleop and endTeleop are camel case functions and all your other functions are underscored.

bgrayd committed:
Fixed typos found in review

bgrayd committed:
I had forgotten to take out a function

This file originally came from somewhere else that needed to call a function on exit.  I removed to the setting up of the call but forgot the function

bgrayd committed:
Added first set of service calls

bgrayd committed:
Added other services and fixed function name formatting

bgrayd commented:
I had not seen the comment about the function names, that has been fixed.  I also added in the service calls.
